include/crocoddyl/multibody/actions/constraint-fwddyn.hxx

```diff
@@ -0,0 +1,196 @@
+      d->pinocchio.ddq_dtau, d->pinocchio.dlambda_dq, d->pinocchio.dlambda_dv, d->pinocchio.dlambda_dtau);
+
+  d->Fu.noalias() = d->pinocchio.ddq_dtau * d->multibody.actuation->dtau_du;
+  costs_->calcDiff(d->costs, x, u);
+}
+
+template <typename Scalar>
+boost::shared_ptr<DifferentialActionDataAbstractTpl<Scalar> >
+DifferentialActionModelConstraintFwdDynamicsTpl<Scalar>::createData() {
+  boost::shared_ptr<DifferentialActionDataAbstractTpl<Scalar> > data =
+      boost::make_shared<DifferentialActionDataConstraintFwdDynamics>(this);
+  DifferentialActionDataConstraintFwdDynamics* d =
+      static_cast<DifferentialActionDataConstraintFwdDynamics*>(data.get());
+  pinocchio::initConstraintDynamics(pinocchio_, d->pinocchio, contacts_);
+
+  return data;
+}
+
+template <typename Scalar>
+bool DifferentialActionModelConstraintFwdDynamicsTpl<Scalar>::checkData(
+    const boost::shared_ptr<DifferentialActionDataAbstract>& data) {
+  boost::shared_ptr<DifferentialActionDataConstraintFwdDynamics> d =
+      boost::dynamic_pointer_cast<DifferentialActionDataConstraintFwdDynamics>(data);
+  if (d != NULL) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+/*
+template <typename Scalar>
+void DifferentialActionModelConstraintFwdDynamicsTpl<Scalar>::quasiStatic(const
+boost::shared_ptr<DifferentialActionDataAbstract>& data, Eigen::Ref<VectorXs> u, const Eigen::Ref<const VectorXs>& x,
+                                                                     const std::size_t& maxiter, const Scalar& tol) {
+if (static_cast<std::size_t>(u.size()) != nu_) {
+  throw_pretty("Invalid argument: "
+               << "u has wrong dimension (it should be " + std::to_string(nu_) + ")");
+}
+if (static_cast<std::size_t>(x.size()) != state_->get_nx()) {
+  throw_pretty("Invalid argument: "
+               << "x has wrong dimension (it should be " + std::to_string(state_->get_nx()) + ")");
+}
+// Static casting the data
+DifferentialActionDataConstraintFwdDynamicsTpl<Scalar>* d =
+    static_cast<DifferentialActionDataConstraintFwdDynamicsTpl<Scalar>*>(data.get());
+const Eigen::VectorBlock<const Eigen::Ref<const VectorXs>, Eigen::Dynamic> q = x.head(state_->get_nq());
+const Eigen::VectorBlock<const Eigen::Ref<const VectorXs>, Eigen::Dynamic> v = x.tail(state_->get_nv());
+
+d->multibody.actuation->tau = pinocchio::rnea(pinocchio_, d->pinocchio, q, v, VectorXs::Zero(state_->get_nv()));
+actuation_->get_actuated(d->multibody.actuation, u);
+}
+*/
+
+template <typename Scalar>
+pinocchio::ModelTpl<Scalar>& DifferentialActionModelConstraintFwdDynamicsTpl<Scalar>::get_pinocchio() const {
+  return pinocchio_;
+}
+
+template <typename Scalar>
+const boost::shared_ptr<ActuationModelAbstractTpl<Scalar> >&
+DifferentialActionModelConstraintFwdDynamicsTpl<Scalar>::get_actuation() const {
+  return actuation_;
+}
+
+template <typename Scalar>
+const std::vector<pinocchio::RigidConstraintModelTpl<Scalar, 0>,
```

Comment:
It will imply memory allocation. You can change the active contacts by providing the vector of active contacts.
We can think to change, but this not the priority for now.